runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -360,23 +360,39 @@ function M.is_enabled(bufnr)
   return bufstates[bufnr] and bufstates[bufnr].enabled or false
 end
 
+--- Optional filters |kwargs|, or `nil` for all.
+--- @class vim.lsp.inlay_hint.enable.Filter
+--- @inlinedoc
+--- Buffer number, or 0/nil for current buffer.
+--- @field bufnr integer?
+
 --- Enables or disables inlay hints for a buffer.
 ---
 --- To "toggle", pass the inverse of `is_enabled()`:
 ---
 --- ```lua
---- vim.lsp.inlay_hint.enable(0, not vim.lsp.inlay_hint.is_enabled())
+--- vim.lsp.inlay_hint.enable(not vim.lsp.inlay_hint.is_enabled())
 --- ```
 ---
---- @param bufnr (integer|nil) Buffer handle, or 0 or nil for current
 --- @param enable (boolean|nil) true/nil to enable, false to disable
+--- @param filter vim.lsp.inlay_hint.enable.Filter?
 --- @since 12
-function M.enable(bufnr, enable)
-  vim.validate({ enable = { enable, 'boolean', true }, bufnr = { bufnr, 'number', true } })
+function M.enable(enable, filter)
+  if type(enable) == 'number' or type(filter) == 'boolean' then
+    vim.deprecate(
+      'vim.lsp.inlay_hint.enable(bufnr:number, enable:boolean)',
+      'vim.diagnostic.enable(enable:boolean, filter:table)',
+      '0.10-dev'
```

Comment:
I don't even know what this is supposed to convey. We can remove this function right now according to this.

Reply:
This is using `vim.deprecate()` to show a useful message to HEAD users. Instead of having to build a similar message manually.

Reply:
I was referring to the version number. Currently this will say `This feature will be removed in Nvim version 0.10-dev`, but since we're already in `0.10-dev` we could just remove it right now. I suspect the provided version should be `0.11`, at which case we can remove it when we're in `0.11-dev` (i.e. after `0.10` is released).

Reply:
No, it will say "this feature _was removed_", see https://github.com/neovim/neovim/pull/28374/files#r1569168989



> but since we're already in `0.10-dev` we could just remove it right now

I'm quite aware of that. Again, I want to use `vim.deprecate` to show a useful message for HEAD users.

Reply:
OK, this was a misunderstanding from my part. I didn't know the function had been adjusted to deal with past removals.